iroh-rpc-types/proto/store.proto

```diff
@@ -0,0 +1,43 @@
+syntax = "proto3";
+
+package store;
+
+service Store {
+  rpc Put(PutRequest) returns (PutResponse) {}
+  rpc Get(GetRequest) returns (GetResponse) {}
+  rpc GetLinks(GetLinksRequest) returns(GetLinksResponse) {}
+}
+
+message PutRequest {
+  // Serialized CID of the given block.
+  bytes cid = 1;
+  // blob of data
+  bytes blob = 2;
+  // list of CIDs
+  repeated bytes links = 3;
+}
+
+// TODO: is there a better way to do an Ack?
+message PutResponse {}
+
+message GetRequest {
+  // Serialized CID of the requested block.
+  bytes cid = 1;
+}
+
+message GetResponse {
+  // TODO: best way to represent a pinnable slice?
+  optional bytes data = 1;
+}
+
+message GetLinksRequest {
+  // Serialized root CID of the requested links.
+  bytes cid = 1;
+}
+
+message GetLinksResponse {
+  // list of CIDs
+  optional repeated bytes links = 1;
+}
+
```

Comment:
Can I get a sanity check here, @dignifiedquire 

Both on if these are the correct methods we want the client to have for the store & if the protobufs are defined correctly.

1) If the expected response is the unit time, do we need to define a protobuf response message?
2) Is it safe to assume that all blobs of data should be returned as `bytes::Bytes`? Specifically looking at the `GetRequest`, the type was a `DBPinnableSlice`.
3) Is there anything we are missing? Do we want an rpc call to shutdown the store process?

Reply:
1. Good question, I actually don't know, can you do some research on how grpc usually handles this?
2. Pinning only works on the same process, so you can treat it as just `Bytes` for the rpc purpose
3. We will want to add lifecycle control methods for iroh-ctl, but I would suggest doing those when you actually need them